Frontend/ui-customer/src/Screens/ParcelDeliveryFlow/ParcelDeliveryScreen/Controller.purs

```diff
@@ -127,24 +128,10 @@ eval (DeliveryDetailAction (PopUpModalController.PersonName (PrimaryEditTextCont
   continue $ state { props { editDetails { name = newVal }}}
 
 eval (DeliveryDetailAction (PopUpModalController.PersonName PrimaryEditTextController.TextImageClicked)) state = do
-  continueWithCmd state $ [do
-    isPickContact <- runEffectFn1 JB.pickContact ""
-    if not isPickContact then
-      void $ pure $ JB.toast "Unable to pick contact, update app"
-    else
-      void $ pure unit
-    pure NoAction
-    ]
+  updateAndExit state $ GoToSelectContact state
 
 eval (DeliveryDetailAction (PopUpModalController.PersonMobile PrimaryEditTextController.TextImageClicked)) state = do
-  continueWithCmd state $ [do
-    isPickContact <- runEffectFn1 JB.pickContact ""
-    if not isPickContact then
-      void $ pure $ JB.toast "Unable to pick contact, update app"
-    else
-      void $ pure unit
-    pure NoAction
-  ]
+  updateAndExit state $ GoToSelectContact state
 
```

Comment:
@ashishsingh101 please have this line added in the controller as there is an edge case when
someone selects a contact and edit the text and then again selects the same contact then the updated value is not getting rendered
`
eval (DeliveryDetailAction (PopUpModalController.PersonName PrimaryEditTextController.TextImageClicked)) state = do
  let updatedState = if state.data.currentStage == ST.SENDER_DETAILS then state {data { senderDetails { name = state.props.editDetails.name, phone = state.props.editDetails.phone}}} else state {data { receiverDetails { name = state.props.editDetails.name, phone = state.props.editDetails.phone}}}
  updateAndExit updatedState $ GoToSelectContact updatedState

eval (DeliveryDetailAction (PopUpModalController.PersonMobile PrimaryEditTextController.TextImageClicked)) state = do
  let updatedState = if state.data.currentStage == ST.SENDER_DETAILS then state {data { senderDetails { name = state.props.editDetails.name, phone = state.props.editDetails.phone}}} else state {data { receiverDetails { name = state.props.editDetails.name, phone = state.props.editDetails.phone}}}
  updateAndExit updatedState $ GoToSelectContact updatedState
`